quickwit-search/src/client.rs

```diff
@@ -28,11 +29,26 @@ use tokio_stream::wrappers::UnboundedReceiverStream;
 use tonic::transport::Channel;
 use tonic::transport::Endpoint;
 use tonic::Request;
+use tracing_opentelemetry::OpenTelemetrySpanExt;
+use tracing::*;
 
 use crate::error::parse_grpc_error;
 use crate::SearchError;
 use crate::SearchService;
 
+struct MetadataMap<'a>(&'a mut tonic::metadata::MetadataMap);
+
+impl<'a> Injector for MetadataMap<'a> {
+    /// Set a key and value in the MetadataMap.  Does nothing if the key or value are not valid inputs
+    fn set(&mut self, key: &str, value: String) {
```

Comment:
`value` could be a `&str`

Reply:
actually the trait `Injector` expects a `String`.